site-scrapers/Atrius.js

```diff
@@ -14,134 +15,31 @@ module.exports = async function GetAvailableAppointments() {
-        hostname: 'myhealth.atriushealth.org',
-        port: 443,
-        path: '/OpenScheduling/OpenScheduling/GetScheduleDays',
-        method: 'POST',
-        headers: {
-            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-            'Content-Length': postData.length,
-            __RequestVerificationToken: verificationToken,
-            Cookie: cookie,
-        },
-    };
-
-    // now request the schedule
-    const schedulePromise = new Promise(resolve => {
-        const req = https.request(options, res => {
-            let body = '';
-            responseHeaders = res.headers;
-            res.on('data', chunk => {
-                body += chunk;
-            });
-            res.on('end', () => {
-                if (res.statusCode >= 200 && res.statusCode < 300) {
-                    resolve(JSON.parse(body));
-                } else {
-                    console.error('Error status code returned from ' + res.url);
-                    resolve();
-                }
-            });
-        });
-        req.write(postData);
-        req.on('error', e => {
-            console.error('Error making scheduling request : ' + e);
-        });
-        req.end();
-    });
 
-    const postResponse = await schedulePromise;
+    // Setup the return object.
     const results = { availability: {}, hasAvailability: false };
-    if (postResponse) {
-        allDays = postResponse.AllDays;
-
-        // The AllDays property is an object, not an array.
-        // The property of the object is generated from the date, which we don't know
-        // We need to iterate over all the properties to determine the dates and the slots.
-        for (const dayEntryProperty in allDays) {
-            if (
-                Object.prototype.hasOwnProperty.call(allDays, dayEntryProperty)
-            ) {
-                dayEntry = allDays[dayEntryProperty];
-                const date = new Date(dayEntry.DateISO);
-                const slots = dayEntry.Slots;
-                const slotsAvailable = slots.length || 0;
-
-                if (date >= new Date()) {
-                    results.availability[
-                        `${date.getMonth() + 1}/${date.getDate() +
-                            1}/${date.getFullYear()}`
-                    ] = {
-                        numberAvailableAppointments: slotsAvailable,
-                        hasAvailability: !!slotsAvailable,
-                    };
-                    if (slotsAvailable) {
-                        results.hasAvailability = true;
-                    }
-                }
-            }
-        }
-    } else {
-        console.error('Null response returned from scheduling request');
-    }
+    return mychart.AddFutureWeeks(
+        "myhealth.atriushealth.org",
+        "/OpenScheduling/OpenScheduling/GetScheduleDays",
+        results,
+        cookie,
+        verificationToken,
+        10,
+        PostDataCallback
+    );
     return results;
```

Comment:
2 return statements here.
Also, I don't love us modifying `results` as a side-effect of AddFutureWeeks - could we instead just return an object with the same structure as `results` instead of passing it in?

Reply:
I changed AddFutureWEeks to return an results object instead of accepting as an argument